crates/gitbutler-testsupport/src/testing_repository.rs

```diff
@@ -94,16 +108,60 @@ pub fn assert_tree_matches<'a>(
     tree: &git2::Tree<'a>,
     files: &[(&str, &[u8])],
 ) {
-    for (path, content) in files {
-        let blob = tree.get_path(Path::new(path)).unwrap().id();
-        let blob: git2::Blob<'a> = repository.find_blob(blob).unwrap();
+    assert_tree_matches_with_mode(
+        repository,
+        tree.id(),
+        &files
+            .iter()
+            .map(|(path, content)| (*path, *content, &[] as &[EntryAttribute]))
+            .collect::<Vec<_>>(),
+    );
+}
+
+#[derive(Debug, Copy, Clone)]
+pub enum EntryAttribute {
+    Tree,
+    Commit,
+    Link,
+    Blob,
+    Executable,
+}
+
+pub fn assert_tree_matches_with_mode(
+    repository: &git2::Repository,
+    tree: git2::Oid,
+    files: &[(&str, &[u8], &[EntryAttribute])],
+) {
+    let gix_repository = gix::open(repository.path()).unwrap();
```

Comment:
It's better for tests (and faster) to open like this: `gix::open_opts(path, gix::open::Options::isolated())`.